packages/generator-rest/src/generator.ts

```diff
@@ -13,71 +13,128 @@ import {
-        '-i',
-        pathToAdjustedOpenApiDefFile,
-        '-g',
-        'typescript-axios',
-        '-o',
-        folderForService,
-        '-t',
-        pathToTemplates,
-        '--api-package',
-        'api',
-        '--model-package',
-        'model',
-        '--config',
-        pathToMustacheValues,
-        '--skip-validate-spec'
-      ];
+  const project = new Project(projectOptions());
+
+  const openApiServiceMetadata = await Promise.all(
+    files.map(async file =>
+      generateOneApi(file, options, pathToTemplates, pathToMustacheValues)
+    )
+  );
+  openApiServiceMetadata.map(metadata =>
+    generateSourcesForService(metadata, project, options)
+  );
+  await project.save();
+}
 
-      logger.info(`Argument for openapi generator ${generationArguments}`);
-
-      try {
-        const response = await execa('npx', generationArguments).catch(err => {
-          logger.error('In exception block 2');
-          logger.error(err);
-        });
-        if (response !== undefined) {
-          logger.info(`Generated the client ${response.stdout}`);
-        }
-      } catch (err) {
-        logger.error('In exception block');
-        logger.error(err);
-      }
-    })
+async function generateOneApi(
+  file: string,
+  options: GeneratorOptions,
+  pathToTemplates: string,
+  pathToMustacheValues: string
+) {
+  const folderForService = getFolderForService(options.outputDir, file);
+  mkdirSync(folderForService);
+
+  const serviceName = getServiceNameCamelCase(file);
+  const adjustedOpenApiContent = getAdjustedOpenApiFile(
+    join(options.inputDir, file),
+    serviceName
+  );
+  const pathToAdjustedOpenApiDefFile = join(folderForService, 'open-api.json');
+  writeJsonSync(pathToAdjustedOpenApiDefFile, adjustedOpenApiContent);
+  // todo
+  await generateFilesUsingOpenAPI(
+    folderForService,
+    pathToAdjustedOpenApiDefFile,
+    pathToTemplates,
+    pathToMustacheValues
   );
+
+  const openApiModel = toOpenApiModel(
+    pathToAdjustedOpenApiDefFile,
+    serviceName,
+    folderForService
+  );
+  return openApiModel;
+}
+
+function generateSourcesForService(
+  serviceMetadata: OpenApiServiceMetadata,
+  project: Project,
+  options: GeneratorOptions
+) {
+  const serviceDir = project.createDirectory(
+    resolve(options.outputDir.toString(), serviceMetadata.serviceDir)
+  );
+  logger.info(`Generating request builder in ${serviceDir}.`);
+  sourceFile(
+    serviceDir,
+    'request-builder',
+    requestBuilderSourceFile(serviceMetadata),
+    true
+  );
+}
+
+async function generateFilesUsingOpenAPI(
+  folderForService: string,
+  pathToAdjustedOpenApiDefFile: string,
+  pathToTemplates: string,
+  pathToMustacheValues: string
+) {
+  const generationArguments = [
+    'openapi-generator',
+    'generate',
+    '-i',
+    pathToAdjustedOpenApiDefFile,
+    '-g',
+    'typescript-axios',
+    '-o',
+    folderForService,
+    '-t',
+    pathToTemplates,
+    '--api-package',
+    'api',
```

Comment:
I wonder why there is no`mode` and `api` folder in the generated code. I find it a bit easier to investigate if they are that is why I set them. But I also remember that under certain conditions (perhaps with the templates or so) they were ignored.

Reply:
https://github.wdf.sap.corp/MA/scp-rocket/issues/1524